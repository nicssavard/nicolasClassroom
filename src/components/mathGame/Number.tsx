interface Props {
  number: number;
  guess: (guess: number) => void;
}

export default function Number({ number, guess }: Props) {
  return (
    <div
      data-testid={`${number}`}
      className="m-1 rounded-lg bg-palette-700 p-1 shadow duration-300 hover:scale-105 hover:bg-palette-800 sm:p-2"
      onClick={() => guess(number)}
    >
      <div>
        <div className="relative flex h-16 w-16 justify-center text-6xl text-gold-500 sm:h-32 sm:w-32 lg:h-40 lg:w-40 1080:h-64 1080:w-64">
          <div className="my-auto">{number}</div>
        </div>
      </div>
    </div>
  );
}
