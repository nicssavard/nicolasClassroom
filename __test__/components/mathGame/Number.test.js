import renderer from "react-test-renderer";
import Number from "../../../src/components/mathGame/Number";
import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";

// test("renders a number", () => {
//   const component = renderer.create(<Number number={1} />);
//   let tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });

test("id", () => {
  const number = 1;
  render(<Number number={number} />);
  expect(screen.getByTestId(`${number}`)).toHaveTextContent(`${number}`);
});
