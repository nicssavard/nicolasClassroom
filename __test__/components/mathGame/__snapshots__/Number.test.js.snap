// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a number 1`] = `
<div
  className="number"
>
  1
</div>
`;
